### PR TITLE
add comment in several hack/ sh scripts.

### PR DESCRIPTION
**Pull request by KobayashiD27**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
/kind cleanup

**What this PR does / why we need it**:
update the comment of them
> kubernetes/hack/benchmark-go.sh
kubernetes/hack/build-cross.sh
kubernetes/hack/build-go.sh
kubernetes/hack/cherry_pick_pull.sh
kubernetes/hack/dev-build-and-push.sh
kubernetes/hack/dev-build-and-up.sh
kubernetes/hack/dev-push-conformance.sh
kubernetes/hack/e2e-node-test.sh
kubernetes/hack/generate-docs.sh

**Which issue(s) this PR fixes**:

Ref #86597 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: hack/build-go.sh [hunk 1]
```diff
@@ -16,6 +16,10 @@
 
 # This script is a vestigial redirection.  Please do not add "real" logic.
 
+# This script runs `make all` command.
+# The command conpiles all Makefile configs.
```
Comment (SataQiu) [inline hack/build-go.sh]: `conpiles` ? Spelling mistake ?

Comment (KobayashiD27) [reply hack/build-go.sh]: Thank you for your review. I will fix it.

File: hack/build-go.sh [hunk 2]
```diff
@@ -16,6 +16,10 @@
 
 # This script is a vestigial redirection.  Please do not add "real" logic.
 
+# This script runs `make all` command.
+# The command compiles all Makefile configs.
+# Usage: `hack/build-go.sh`.
```
Comment (spiffxp) [inline hack/build-go.sh]: I would call out that this uses the WHAT env var and provide an additional example

File: hack/benchmark-go.sh [hunk 3]
```diff
@@ -14,6 +14,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# This script runs `make test` command with some args for benchmark test.
+# The "true" target of this makerule is `hack/make-rules/test.sh`.
+# Usage: `hack/benchmark-go.sh`.
```
Comment (spiffxp) [inline hack/benchmark-go.sh]: I would call out that this uses the WHAT env var and provide an additional example

File: hack/e2e-node-test.sh [hunk 4]
```diff
@@ -15,6 +15,11 @@
 # limitations under the License.
 
 # This script is a vestigial redirection.  Please do not add "real" logic.
+# The "true" target of this makerule is `hack/make-rules/test-e2e-node.sh`.
+
+# This script runs `make test-e2e-node` command.
+# The command builds and runs node end-to-end tests.
+# Usage: `hack/e2e-node-test.sh`.
```
Comment (spiffxp) [inline hack/e2e-node-test.sh]: I would call out that this uses the FOCUS and SKIP env vars and provide an additional example

**Review by spiffxp — COMMENTED** [review]
Thanks for the docs, some tiny nits

**Comment by KobayashiD27** [comment]
> Thanks for the docs, some tiny nits

Thank you for your reviews. I fix these comments.

**Pull request merged**